diff --git a/usr/share/pve-manager/js/apidoc.js b/usr/share/pve-manager/js/apidoc.js
index 3aaf09f..0a34b6e 100644
--- a/usr/share/pve-manager/js/apidoc.js
+++ b/usr/share/pve-manager/js/apidoc.js
@@ -50336,6 +50336,37 @@
                            "type" : "string",
                            "typetext" : "<string>"
                         },
+                        "truenas_user" : {
+                           "description" : "TrueNAS user for API access",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
+                        "truenas_password" : {
+                           "description" : "TrueNAS password for API access (Deprecated)",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
+                        "truenas_secret" : {
+                           "description" : "TrueNAS Secret for API access",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
+                        "truenas_use_ssl" : {
+                           "description" : "TrueNAS API access via SSL",
+                           "optional" : 1,
+                           "type" : "boolean",
+                           "typetext" : "<boolean>"
+                        },
+                        "truenas_apiv4_host" : {
+                           "description" : "TrueNAS API Host via IPv4",
+                           "format" : "address",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
                         "fuse" : {
                            "description" : "Mount CephFS through FUSE.",
                            "optional" : 1,
@@ -50555,6 +50586,12 @@
                            "type" : "boolean",
                            "typetext" : "<boolean>"
                         },
+                        "target" : {
+                           "description" : "iSCSI target.",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
                         "transport" : {
                            "description" : "Gluster transport: tcp or rdma",
                            "enum" : [
@@ -50854,6 +50891,37 @@
                      "optional" : 1,
                      "type" : "string",
                      "typetext" : "<string>"
+                  },
+                  "truenas_user" : {
+                     "description" : "TrueNAS user for API access",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
+                  },
+                  "truenas_password" : {
+                     "description" : "TrueNAS password for API access (Deprecated)",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
+                  },
+                  "truenas_secret" : {
+                     "description" : "TrueNAS secret for API access",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
+                  },
+                  "truenas_use_ssl" : {
+                     "description" : "TrueNAS API access via SSL",
+                     "optional" : 1,
+                     "type" : "boolean",
+                     "typetext" : "<boolean>"
+                  },
+                  "truenas_apiv4_host" : {
+                     "description" : "TrueNAS API Host via IPv4",
+                     "format" : "address",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
                   },
                   "fuse" : {
                      "description" : "Mount CephFS through FUSE.",
