--- pvemanagerlib-5.2-9.js	2018-10-16 07:17:21.915593478 -0400
+++ pvemanagerlib.js	2018-10-21 21:52:20.781740237 -0400
@@ -5597,6 +5597,7 @@
     alias: ['widget.pveiScsiProviderSelector'],
     comboItems: [
 	['comstar', 'Comstar'],
+	['truenas', 'TrueNAS-API'],
 	[ 'istgt', 'istgt'],
 	[ 'iet', 'IET'],
 	[ 'LIO', 'LIO']
@@ -28574,6 +28575,7 @@
 	data: {
 	    isLIO: false,
 	    isComstar: true,
+	    isTrueNAS: false,
 	    hasWriteCacheOption: true
 	}
     },
@@ -28586,10 +28588,19 @@
 	    }
 	},
 	changeISCSIProvider: function(f, newVal, oldVal) {
+	    var me = this;
 	    var vm = this.getViewModel();
 	    vm.set('isLIO', newVal === 'LIO');
 	    vm.set('isComstar', newVal === 'comstar');
-	    vm.set('hasWriteCacheOption', newVal === 'comstar' || newVal === 'istgt');
+	    vm.set('isTrueNAS', newVal === 'truenas');
+	    vm.set('hasWriteCacheOption', newVal === 'comstar' || newVal === 'truenas' || newVal === 'istgt');
+	    if (newVal !== 'truenas') {
+		me.lookupReference('truenas_use_ssl_field').setValue(false);
+		me.lookupReference('truenas_apiv4_host_field').setValue('');
+		me.lookupReference('truenas_user_field').setValue('');
+		me.lookupReference('truenas_password_field').setValue('');
+		me.lookupReference('truenas_confirmpw_field').setValue('');
+	    }
 	}
     },
 
@@ -28607,6 +28618,7 @@
     },
 
     setValues: function diff(values) {
+	values.truenas_confirmpw = values.truenas_password;
 	values.writecache = values.nowritecache ? 0 : 1;
 	this.callParent([values]);
     },
@@ -28623,7 +28635,7 @@
 		allowBlank: false
 	    },
 	    {
-		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		xtype: 'textfield',
 		name: 'pool',
 		value: '',
 		fieldLabel: gettext('Pool'),
@@ -28637,7 +28649,7 @@
 		allowBlank: false
 	    },
 	    {
-		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		xtype: 'textfield',
 		name: 'target',
 		value: '',
 		fieldLabel: gettext('Target'),
@@ -28648,8 +28660,28 @@
 		name: 'comstar_tg',
 		value: '',
 		fieldLabel: gettext('Target group'),
-		bind: me.isCreate ? { disabled: '{!isComstar}' } : { hidden: '{!isComstar}' },
+		bind: { hidden: '{!isComstar}' },
 		allowBlank: true
+	    },
+	    {
+		xtype: 'proxmoxcheckbox',
+		name: 'truenas_use_ssl',
+		reference: 'truenas_use_ssl_field',
+		inputId: 'truenas_use_ssl_field',
+		checked: false,
+		bind: { hidden: '{!isTrueNAS}' },
+		uncheckedValue: 0,
+		fieldLabel: gettext('API use SSL')
+	    },
+	    {
+		xtype: 'textfield',
+		name: 'truenas_user',
+		reference: 'truenas_user_field',
+		inputId: 'truenas_user_field',
+		value: '',
+		allowBlank: false,
+		fieldLabel: gettext('API Username'),
+		bind: { hidden: '{!isTrueNAS}' },
 	    }
 	];
 
@@ -28680,7 +28712,7 @@
 		xtype: me.isCreate ? 'textfield' : 'displayfield',
 		name: 'comstar_hg',
 		value: '',
-		bind: me.isCreate ? { disabled: '{!isComstar}' } : { hidden: '{!isComstar}' },
+		bind: { hidden: '{!isComstar}' },
 		fieldLabel: gettext('Host group'),
 		allowBlank: true
 	    },
@@ -28688,9 +28720,61 @@
 		xtype: me.isCreate ? 'textfield' : 'displayfield',
 		name: 'lio_tpg',
 		value: '',
-		bind: me.isCreate ? { disabled: '{!isLIO}' } : { hidden: '{!isLIO}' },
-		allowBlank: false,
-		fieldLabel: gettext('Target portal group')
+		bind: { hidden: '{!isLIO}' },
+		fieldLabel: gettext('Target portal group'),
+		allowBlank: true
+	    },
+	    {
+		xtype: 'proxmoxtextfield',
+		name: 'truenas_apiv4_host',
+		reference: 'truenas_apiv4_host_field',
+		value: '',
+                editable: true,
+                deleteEmpty: true,
+                allowBlank: true,
+                emptyText: Proxmox.Utils.noneText,
+		bind: { hidden: '{!isTrueNAS}' },
+		fieldLabel: gettext('API IPv4 Host'),
+		allowBlank: true
+	    },
+	    {
+		xtype: 'proxmoxtextfield',
+		name: 'truenas_password',
+		reference: 'truenas_password_field',
+                inputType: me.isCreate ? '' : 'password',
+                value: '',
+                editable: true,
+                deleteEmpty: true,
+                allowBlank: false,
+                emptyText: Proxmox.Utils.noneText,
+		bind: { hidden: '{!isTrueNAS}' },
+		fieldLabel: gettext('API Password'),
+                change: function(f, value) {
+                    if (f.rendered) {
+                        f.up().down('field[name=truenas_confirmpw]').validate();
+                    }
+                }
+	    },
+	    {
+		xtype: 'proxmoxtextfield',
+		name: 'truenas_confirmpw',
+		reference: 'truenas_confirmpw_field',
+                inputType: me.isCreate ? '' : 'password',
+                value: '',
+                editable: true,
+                deleteEmpty: true,
+                allowBlank: false,
+		submitValue: false,
+                emptyText: Proxmox.Utils.noneText,
+		bind: { hidden: '{!isTrueNAS}' },
+		fieldLabel: gettext('Confirm Password'),
+                validator: function(value) {
+                    var pw = this.up().down('field[name=truenas_password]').getValue();
+                    if (pw !== value) {
+                        return "Passwords do not match!";
+                    }
+                    return true;
+                }
 	    }
 	];
 
